HAPNodeJSClient.js

```diff
@@ -234,11 +237,13 @@ HAPNodeJSClient.prototype.HAPcontrolByDeviceID = function(deviceID, body, callba
  *
  * @param  {type} ipAddress IP Address of homebridge instance
  * @param  {type} port      Port of homebridge instance
+ * @param  {type} pin       Pin of homebridge instance
  * @param  {type} body      An array of HomeKit characteristic updates, [{ \"aid\": 2, \"iid\": 9, \"value\": 0}]
  * @param  {type} callback  Callback to execute upon completion of characteristic setting, function(err, response)
  */
 
-HAPNodeJSClient.prototype.HAPcontrol = function(ipAddress, port, body, callback) {
+HAPNodeJSClient.prototype.HAPcontrol = function(ipAddress, port, pin, body, callback) {
```

Comment:
This is a breaking change to the external API and can not be merged as it will break all other packages using this package

Reply:
@NorthernMan54 I know. What about making it the last argument, after `callback`?

Reply:
I think It makes more sense though to be able to specify the pin on a per-call basis instead of globally, but leaving a default 'global' pin is OK.

Reply:
Thinking the UX thru on this for a bit.

Alexa - Just add a config object called pins [{deviceId: xx.xx.xx.xx.xx, pin: "031-45-154"}, { deviceId: yy.yy.yy.yy.yy, pin: "031-45-155"}]

Node-Red -  How to capture the pin for each instance ? Could create a config node for each pin.  Should the config node be smart and only show Homebridge devices from the Homebridge instance?  Or keep the existing behaviour of one list for all homebridge accessories.     Personally I prefer the one list approach as I have so many instances floating around, keep losing track of them

Reply:
> Could create a config node for each pin.

You already have this instance though? So, I already did this actually. But in order to use it, we need the code changes I made already or some variant :)

But yes, this is how I do it. For each node, the user can select the corresponding config node for it. It's a bit tedious but it does work!

> only show Homebridge devices from the Homebridge instance

This makes a ton of sense to me. The only problem is, how would we do it? In insecure mode, all accessories are returned with any possible pin#.

If it's possible to split by pin#, I think it makes more sense to do so vs. one list. While the one list approach is convenient, at *some* point in the config, the user will still have to input the pin# so it sort of defeats the purpose.

However, if it *IS* possible to know which devices in the list belong to which pin# then it would be really amazing not to even have to select it and have the config nodes take care of it. E.g.

- Suppose user has 5 hb instances, but only adds 2 of them to NR
- These 2 have PINS that end in -4 and -5 respectively
- NR presents user a list of all devices only for these pin#s
- NR hides all other devices (for which, we don't know the pin#)

This is a *bit* tricky still though, for ex., in the case that user adds a 3rd hb instance to NR, and this 3rd hb instance has same pin# as the first.

Reply:
Internally, I'm transitioning from using host:port as the unique identifier of a homebridge instance to using the "username" field from the config.json.  This is how HomeKit identifies each instance.   Has been working with the Alexa beta for a few months now.

The config node, in multi pin mode could have two input fields, the username and pin

But is this potentially over complicating this?

Reply:
> Internally, I'm transitioning from using host:port as the unique identifier of a homebridge instance to using the "username" field from the config.json. This is how HomeKit identifies each instance. Has been working with the Alexa beta for a few months now.

Great idea. So, `username` is a truly unique identifier for hb instances? (e.g, port doesn't matter?)

> The config node, in multi pin mode could have two input fields, the username and pin

I see. So, you *could* separate the pin#'s because you have different usernames/deviceIds.

This approach you've outlined of username/pin in the config nodes makes a lot of sense to me. Maybe we should add one more, just a way for a user to assign a custom name to keep track of the instances so they don't have to rely on deviceId or PIN.

I like the idea of including a pin number per-username/deviceId, because it's practical, logical, and there's no way to guess the user's pin and there can be multiple different pins. In fact, I didn't realize that hb can tolerate identical pins for multiple instances...